test/v1/controllers/validators/CreateAmendCgtPpdOverridesValidatorFactorySpec.scala

```diff
@@ -551,193 +555,301 @@ class CreateAmendCgtPpdOverridesValidatorSpec extends UnitSpec with ValueFormatE
-  private val bothGainsAndLossSinglePropertyDisposalsRequestBody     = AnyContentAsJson(bothGainsAndLossSinglePropertyDisposalsRequestBodyJson)
-  private val neitherGainsOrLossMultiplePropertyDisposalsRequestBody = AnyContentAsJson(neitherGainsOrLossMultiplePropertyDisposalsRequestBodyJson)
-  private val neitherGainsOrLossSinglePropertyDisposalsRequestBody   = AnyContentAsJson(neitherGainsOrLossSinglePropertyDisposalsRequestBodyJson)
 
-  class Test extends MockAppConfig {
 
-    // implicit val dateTimeProvider: CurrentDateTime = mockCurrentDateTime
-    // val dateTimeFormatter: DateTimeFormatter = DateTimeFormat.forPattern("yyyy-MM-dd")
-    implicit val appConfig: AppConfig        = mockAppConfig
 
-    val validator = new CreateAmendCgtPpdOverridesValidator()
+  private val parsedNino    = Nino(validNino)
+  private val parsedTaxYear = TaxYear.fromMtd(validTaxYear)
+  private val parsedValidRequestBody = validRequestJson.as[CreateAmendCgtPpdOverridesRequestBody]
+  private val parsedValidMultipleOnlyBody = validOnlyMultiplePropertyDisposalsRequestJson.as[CreateAmendCgtPpdOverridesRequestBody]
+  private val parsedValidSingleOnlyBody = validOnlySinglePropertyDisposalsRequestJson.as[CreateAmendCgtPpdOverridesRequestBody]
 
-    //MockCurrentDateTime.getDateTime
-    //  .returns(DateTime.parse("2021-07-11", dateTimeFormatter))
-    //  .anyNumberOfTimes()
+  private val validatorFactory                         = new CreateAmendCgtPpdOverridesValidatorFactory(mockAppConfig)
+  private def validator(nino: String, taxYear: String, body: JsValue) = validatorFactory.validator(nino, taxYear, body)
 
     MockedAppConfig.minimumPermittedTaxYear
       .returns(2020)
       .anyNumberOfTimes()
 
-  }
 
-  "running a validation" should {
-    "return no errors" when {
-      "a valid request is supplied" in new Test {
-        validator.validate(CreateAmendCgtPpdOverridesRawData(validNino, validTaxYear, validRequestBody)) shouldBe Nil
-      }
 
-      "a valid request contains only multiple disposals is supplied" in new Test {
-        validator.validate(
-          CreateAmendCgtPpdOverridesRawData(validNino, validTaxYear, AnyContentAsJson(validOnlyMultiplePropertyDisposalsRequestJson))) shouldBe Nil
+  "validator" should {
+    "return the parsed domain object" when {
+      "a valid request is supplied" in {
+        val result: Either[ErrorWrapper, CreateAmendCgtPpdOverridesRequestData] =
+          validator(validNino, validTaxYear, validRequestJson).validateAndWrapResult()
+        result shouldBe Right(CreateAmendCgtPpdOverridesRequestData(parsedNino, parsedTaxYear,parsedValidRequestBody))
       }
+    }
 
-      "a valid request contains only single disposals is supplied" in new Test {
-        validator.validate(
-          CreateAmendCgtPpdOverridesRawData(validNino, validTaxYear, AnyContentAsJson(validOnlySinglePropertyDisposalsRequestJson))) shouldBe Nil
+      "a valid request contains only multiple disposals is supplied" in  {
```

Comment:
Shouldn't this read: 
```suggestion
      "a valid request containing only multiple disposals is supplied" in  {
```